packages/flutter_tools/test/general.shard/base/os_test.dart

```diff
@@ -88,6 +89,73 @@ void main() {
       expect(result[1].path, kPath2);
     });
   });
+
+  group('makeExecutable', () {
+    Directory tempDir;
+
+    setUp(() {
+      tempDir = globals.fs.systemTempDirectory.createTempSync('flutter_tools_os_utils_test.');
+    });
+
+    tearDown(() {
+      tryToDelete(tempDir);
+    });
+
+    testUsingContext('makeExecutable', () async {
+      final File file = globals.fs.file(globals.fs.path.join(tempDir.path, 'foo.script'));
+      file.writeAsStringSync('hello world');
+      globals.os.makeExecutable(file);
+
+      // Skip this test on windows.
+      if (!globals.platform.isWindows) {
+        final String mode = file.statSync().modeString();
+        // rwxr--r--
+        expect(mode.substring(0, 3), endsWith('x'));
+      }
+    }, overrides: <Type, Generator>{
+      OperatingSystemUtils: () => OperatingSystemUtils(
+        fileSystem: globals.fs,
+        logger: globals.logger,
+        platform: globals.platform,
+        processManager: globals.processManager,
+      ),
+    });
+  });
```

Comment:
maybe: skip: const LocalPlatform().isWindows? Not sure which is better.

Reply:
Done.